src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/PreferHashDataOverComputeHash.Fixer.cs

```diff
@@ -0,0 +1,81 @@
+            var computeHashNode = root.FindNode(context.Span, getInnermostNodeForTie: true);
+            var diagnostics = context.Diagnostics[0];
+            var bufferArgNode = root.FindNode(diagnostics.AdditionalLocations[0].SourceSpan);
+            if (computeHashNode is null || bufferArgNode is null)
+            {
+                return;
+            }
+            var hashTypeName = diagnostics.Properties[PreferHashDataOverComputeHashAnalyzer.TargetHashTypeName];
+
+            switch (diagnostics.AdditionalLocations.Count)
+            {
+                case 1:
+                    {
+                        //chained method SHA256.Create().ComputeHash(buffer)
+                        var codeActionChain = GetCodeAction(context, hashTypeName, bufferArgNode, (editor, hashDataInvoked) =>
+                        {
+                            editor.ReplaceNode(computeHashNode, hashDataInvoked);
+                        });
+                        context.RegisterCodeFix(codeActionChain, diagnostics);
+                        return;
+                    }
+                case 2:
+                    {
+                        var nodeToRemove = root.FindNode(diagnostics.AdditionalLocations[1].SourceSpan);
+                        if (nodeToRemove is null || !TryGetCodeFixer(computeHashNode, bufferArgNode, nodeToRemove, out ApplyCodeFixAction? codeFixer))
+                        {
+                            return;
+                        }
+
+                        context.RegisterCodeFix(GetCodeAction(context, hashTypeName, bufferArgNode, codeFixer), diagnostics);
+                        return;
+                    }
+            }
+        }
+
+        private static CodeAction GetCodeAction(CodeFixContext context, string hashTypeName, SyntaxNode bufferArgNode, ApplyCodeFixAction codeFixer)
+        {
+            return CodeAction.Create(
+                   title: MicrosoftNetCoreAnalyzersResources.PreferHashDataOverComputeHashAnalyzerTitle,
+                   createChangedDocument: async cancellationToken =>
+                   {
+                       DocumentEditor editor = await DocumentEditor.CreateAsync(context.Document, cancellationToken).ConfigureAwait(false);
+                       SyntaxGenerator generator = editor.Generator;
+
+                       // hashTypeName.HashData
+                       var hashData = generator.MemberAccessExpression(generator.IdentifierName(hashTypeName), PreferHashDataOverComputeHashAnalyzer.HashDataMethodName);
+                       var hashDataInvoked = generator.InvocationExpression(hashData, bufferArgNode);
+                       codeFixer(editor, hashDataInvoked);
+
+                       return editor.GetChangedDocument();
+                   },
+                   equivalenceKey: MicrosoftNetCoreAnalyzersResources.PreferHashDataOverComputeHashAnalyzerTitle);
```

Comment:
IIRC, @sharwell said somewhere that equivalenceKey shouldn't be localized. So you should probably use `nameof` here.